terraform/transform_resource.go

```diff
@@ -454,7 +464,16 @@ func (n *graphNodeExpandedResource) managedResourceEvalNodes(resource *Resource,
 					Name:   n.stateId(),
 					Output: &state,
 				},
-
+				// Re-run validation to catch any errors we missed, e.g. type
+				// mismatches on computed values.
+				&EvalValidateResource{
+					Provider:       &provider,
+					Config:         &resourceConfig,
+					ResourceName:   n.Resource.Name,
+					ResourceType:   n.Resource.Type,
+					ResourceMode:   n.Resource.Mode,
+					IgnoreWarnings: true,
+				},
```

Comment:
So this eval sequence is the same for `walkDestroy` and `walkApply`... and I have a sneaking suspicion that we should be skipping this node for `walkDestroy`.

No tests have failed, and I'm not quite sure how to construct a failing test to prove my hunch. It would need to have the properties:

 * Resource passes validation during `walkValidate`
 * But fails validation during `walkDestroy`

Since all our cases so far have involved computed values being resolved, I can't think of any real world example that would trigger this.

So given all that, not sure if I want to blindly add it to `walkDestroy`, or blindly _avoid_ adding it to `walkDestroy`. Thoughts welcome!